src/Bicep.Core/SemanticModel/DeclaredSymbol.cs

```diff
@@ -25,11 +25,13 @@ protected DeclaredSymbol(ISymbolContext context, string name, SyntaxBase declari
         /// <summary>
         /// Gets the syntax node of the identifier.
         /// </summary>
-        public IdentifierSyntax NameSyntax { get; }
+        public IdentifierSyntaxBase NameSyntax { get; }
 
         /// <summary>
         /// Gets the type of the symbol.
         /// </summary>
         public TypeSymbol Type => this.Context.TypeManager.GetTypeInfo(DeclaringSyntax);
+
+        public bool IsIdentifierValid => this.NameSyntax is IdentifierSyntax;
```

Comment:
For O(1) checks I don't see a need to make it a method.

Reply:
+1